hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -827,8 +827,9 @@ protected void putAllInternal(final Map<? extends Object, ? extends Object> entr
         int partitionCount = partitionService.getPartitionCount();
         boolean tooManyEntries = entries.size() > (partitionCount * factor);
         try {
+            List<Future<?>> futures = new LinkedList<Future<?>>();
```

Comment:
I would make use of an ArrayList and initialize it with entries.size. This way you create a constant amount of a litter instead of litter for every item in the entries set.